specs/language/resources.tex

```diff
@@ -663,19 +664,237 @@
+16-byte row. Each row in storage layout (each column for \textit{column\_major}
+matrix) is aligned to a 16-byte row.
+
+\p Members of structured types used in constant buffers follow these same rules.
+Because structures are always 16-byte row-aligned, the offset layout within the
+structure is independent of the particular structure instance location.
+
+\Sub{Packoffset annotations}{Resources.cnbuf.po}
+
+\p Shader constants declared in \texttt{cbuffer} declaration group can have an
+optional \texttt{packoffset} annotation added before their \texttt{;} terminal.
+
+\begin{grammar}
+  \define{packoffset-annotation}\br
+  \terminal{: packoffset(} packoffset-id packoffset-row \opt{packoffset-element} \terminal{)}
+
+  \define{packoffset-id}\textnormal{one of}\br
+  \terminal{c C}
+
+  \define{packoffset-row}\br
+  digit\br
+  packoffset-row digit
+
+  \define{packoffset-element}\br
+  \terminal{.} \textit{packoffset-element-id}
+
+  \define{packoffset-element-id}\textnormal{one of}\br
+  \terminal{x y z w r g b a}\br
+\end{grammar}
+
+\p The \texttt{packoffset} annotation defines specific offset in the constant
+buffer where the shader constant is located.
+
+\p The \textit{packoffset-row} number is the index into the array of 16-byte
+rows of the constant buffer. Each row treated as a vector of four 32-bit
+elements.
+
+\p The \textit{packoffset-element} identifies specific location within the
+vector where \texttt{x} or \texttt{r} is the first element of the vector,
+\texttt{y} or \texttt{g} is the second one, \texttt{z} or \texttt{b} is the
+third one and \texttt{w} or \texttt{a} is the fourth, and last element of the
+vector.
+
+\p Shader constant that has a structure, array or matrix type must always be
+16-byte row-aligned. It means that if it specifies a
+\textit{packoffset-element}, it must have a value of \texttt{x} or \texttt{r}.
+
+\p For example
+
+\begin{HLSL}
+  cbuffer MyConstants {
+    float2 Pos : packoffset(c1.z);
+  };
+\end{HLSL}
+
+The \texttt{Pos} shader constant will be located at byte offset \texttt{24} in
+the constant buffer \texttt{MyConstants}.
+
+\Sub{Default Constant Buffer}{Resources.cnbuf.defcb}
+
+\p All variables declaration in the global scope are implicitly added to default
+constant buffer called \texttt{\$Globals}, unless they are marked
+\texttt{static}, \texttt{groupshared}, or declare a resource or array of
+resources.
+
+\p Layout rules for a default constant buffer are the same as for a named
+\texttt{cbuffer} declaration group.
+
+\p Default constant buffer declarations can have an optional
+\textit{resource-binding} annotation with a \textit{register-type} \texttt{c} or
+\texttt{C} added before their \texttt{;} terminal. This annotation does not
+define a virtual register binding but rather an index into an array of 16-byte
+rows that represents the default constant buffer, similar to the
+\texttt{packoffset} annotation.
+
+\p For example
+\begin{HLSL}
+  float4 CameraPos : register(c2);
```

Comment:
Mostly asking for historical context. Why is it called a register in $Globasl but packoffset for the rest?

Reply:
My guess is there was just a single constant buffer `$Globals` before `cbuffer` blocks and `packoffset` were added, and `register` was an established annotation at that time. So it just got extended with a 'c' value and new meaning.
@tex3d - please correct me if I am wrong! :)